packages/cloudflare/src/handler.ts

```diff
@@ -40,5 +54,52 @@ export function withSentry<E extends ExportedHandler<any>>(
     (handler.fetch as any).__SENTRY_INSTRUMENTED__ = true;
   }
 
+  if (
+    'scheduled' in handler &&
+    typeof handler.scheduled === 'function' &&
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-explicit-any
+    !(handler.scheduled as any).__SENTRY_INSTRUMENTED__
+  ) {
+    handler.scheduled = new Proxy(handler.scheduled, {
+      apply(target, thisArg, args: Parameters<ExportedHandlerScheduledHandler<ExtractEnv<E>>>) {
+        const [event, env, context] = args;
+        return withIsolationScope(isolationScope => {
+          const options = optionsCallback(env);
+          const client = init(options);
+          isolationScope.setClient(client);
+
+          addCloudResourceContext(isolationScope);
+
+          return startSpan(
```

Comment:
This duplication with the request handler is not ideal, but I'd rather refactor this later once I'm more confident in the API design of the `withIsolationScope` callback.